server/ai_live_video.go

```diff
@@ -138,5 +139,35 @@ func startControlPublish(control *url.URL, params aiRequestParams) {
 	}
 	params.node.LiveMu.Lock()
 	defer params.node.LiveMu.Unlock()
-	params.node.LivePipelines[stream] = &core.LivePipeline{ControlPub: controlPub}
+
+	ticker := time.NewTicker(10 * time.Second)
+	done := make(chan bool, 1)
+	stop := func() {
+		ticker.Stop()
+		done <- true
+	}
+
+	params.node.LivePipelines[stream] = &core.LivePipeline{
+		ControlPub:  controlPub,
+		StopControl: stop,
+	}
+
+	// send a keepalive periodically to keep both ends of the connection alive
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				msg := `{"keep":"alive"}`
```

Comment:
**nit:** Would consider a static const

Reply:
const'ified in [533e8a64](https://github.com/livepeer/go-livepeer/pull/3299/commits/533e8a64e8520e42f942dfa6b695675d0aedde13) (kept the declaration local since it's not used anywhere else; presumably the compiler is smart enough to hoist it out of the loop)